test/system/libs/helpers.bash

```diff
@@ -577,3 +577,56 @@ function is_fedora_rawhide() (
 
   return 0
 )
+
+
+# Creates a container with org.freedesktop.Flatpak.SessionHelper D-Bus interface
+#
+# Pulling of an image is taken care of by the function
+#
+# Parameters:
+# ===========
+# - container_name - name of the container
+function create_container_flatpak_session_helper() (
+  local container_name="$1"
+
+  local default_image
+  default_image="$(get_default_image)"
+
+  pull_default_image
+
+  gdbus call \
+    --session \
+    --dest org.freedesktop.Flatpak \
+    --object-path /org/freedesktop/Flatpak/SessionHelper \
+    --method org.freedesktop.Flatpak.SessionHelper.RequestSession
+
+  podman_cmd=(
+      podman create \
+          --dns none \
+          --env TOOLBOX_PATH="$TOOLBX" \
+          --name "$container_name" \
+          --network host \
+          --no-hosts \
+          --pid host \
+          --privileged \
+          --userns=keep-id \
+          --user root:root \
+          --volume "$HOME":"$HOME":rslave \
+          --volume "$TOOLBX":/usr/bin/toolbox \
+          --volume "$XDG_RUNTIME_DIR":"$XDG_RUNTIME_DIR" \
+          --volume "$XDG_RUNTIME_DIR/.flatpak-helper/monitor":/run/host/monitor \
+          --volume /etc:/run/host/etc \
+          --volume /usr:/run/host/usr:rslave \
+          "$default_image" \
+          toolbox init-container \
+                  --home "$HOME" \
+                  --shell "$SHELL" \
+                  --uid "$(id -ru)" \
+                  --user "$USER"
+    )
+
+    # Print the full command
+    # printf "%q " "${podman_cmd[@]}" >> /tmp/toolbox_podman_command.log
+
+    "${podman_cmd[@]}" || fail "Podman couldn't create container '$container_name'"
```

Comment:
Nitpick: we can use `echo` and `return 1`, like you had used when resolving the home directory and parsing the path used by `org.freedesktop.Flatpak.SessionHelper`.

I know `fail` is used in some other parts of the code base.  However, over time, it seems to me that it doesn't have that many benefits over printing a message to the standard error stream and returning with a non zero value.  The `fail` function isn't part of core Bats and comes from [bats-support](https://github.com/bats-core/bats-support), which is sometimes not packaged by distributions, and we have to carry it as a Git submodule in Toolbx.  If we stop using it, then we can drop this dependency.